addons/mass_mailing/tests/common.py

```diff
@@ -141,13 +142,15 @@ def assertMailTraces(self, recipients_info, mailing, records,
                     email, partner, status, record,
                     len(recipient_trace), debug_info)
             )
-            self.assertTrue(bool(recipient_trace.mail_mail_id_int))
+            # In mass_mail mode, messages that would be canceled are not created, only the trace is created.
+            mail_not_created = recipient_trace.trace_status == 'cancel'
```

Comment:
Would do the check we don't have mail.mail / mail.message here, instead of doing it as custom check in tests (continuing previous comment)